### PR TITLE
Decorator

### PR DESCRIPTION
**Pull request by MICHAELMUNAVU83**
- Implement composition (as an example of the design pattern).
- Run a program using the command line.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9]
+    else
+      @nameable.correct_name
+    end
+  end
+end
```
Comment (TedLivist) [inline decorator.rb]: - [ ] **[OPTIONAL]** In keeping with the Ruby convention, each class should be in its own file. I would recommend you move the **CapitalizeDecorator** and **TrimmerDecorator** classes to their own respective files

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @MICHAELMUNAVU83,

✔️ Names are well trimmed and capitalized

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (MICHAELMUNAVU83) [reply decorator.rb]: thanks @TedLivist  , I was trying to do this trial  ...person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name ...


I would appreciate advice on which file I should call it from

Comment (TedLivist) [reply decorator.rb]: Hi @MICHAELMUNAVU83 You can do it in any file and just import the needed classes.

But I would recommend you can create something of a test file and import any needed classes there, and run your trial. This way you don't need to touch the original classes (keeping it clean). And if you don't need the test file anymore you can just delete it.

**Pull request merged**